### ISSUE TITLE
Refactoring: Return non-zero exit codes for error conditions

### ISSUE THREAD

**Issue by stefanprodan** [openfaas/faas-cli#183]
## Expected Behaviour

When a command fails I would expect exit code 1 so automation tools can detect an error occurred and stop/react accordingly. 

## Current Behaviour

Some examples of error handling:

```go
	response, err := proxy.InvokeFunction(gateway, functionName, &functionInput, contentType)
	if err != nil {
		fmt.Println(err)
		return
	}
```

```go
	if len(services.Functions) > 0 {
		pushStack(&services, parallel)
	} else {
		fmt.Println("You must supply a valid YAML file.")
		return
	}
```

```
	if len(services.Functions) > 0 {
		pushStack(&services, parallel)
	} else {
		fmt.Println("You must supply a valid YAML file.")
		return
	}
```

## Possible Solution

* use cobra.Command.RunE instead of Run
* change the signature of all runCmd functions to return error
* propagate errors with pkg/errors and wrap the original error with meaningful messages

**Comment by ericstoekl** [comment]
Derek add label: proposal

**Comment by alexellis** [comment]
Agree with this refactoring. Should ideally return an error. This is tech debt.

**Comment by alexellis** [comment]
I think it would make sense to start with one of the commands in isolation, merging that then doing the others one at a time. Could be easier for bite-size chunks / contributions.

**Comment by nenadilic84** [comment]
I'll work on this if anyone asks 😄

**Comment by alexellis** [comment]
Which one do you want to take first?

**Comment by nenadilic84** [comment]
I'll start with `build` if you're ok with it...

**Comment by nenadilic84** [comment]
Added the PR ^
If it seems ok, I can continue and add other commands as well

**Comment by jonocodes** [comment]
Ok if I take a stab at the list command?

**Comment by alexellis** [comment]
Let's get one finished and merged before doing the next. Yes Jono please do the lists command and reach out on Slack there will be people ready to help and to get you up to speed.

**Comment by alexellis** [comment]
I mean one PR per person, so we have less work in progress and more work delivered.

**Comment by alexellis** [comment]
@stefanprodan @johnmccabe can we compile a quick mark-down table showing what's done/left in terms of non-zero exits? Is faas-cli login/logout covered already?

Thanks. @austinfrey @jockdarock maybe we can direct some of the newer contributors here so they can get their first Go PRs into the project?

**Comment by johnmccabe** [comment]
@alexellis this should probably be closed following the merge of #207

**Comment by rgee0** [comment]
Derek close
